disnake/channel.py

```diff
@@ -1806,6 +1814,491 @@ class NewsChannel(TextChannel):
+        Forbidden
+            You do not have permissions to create a thread.
+        HTTPException
+            Starting the thread failed.
+
+        Returns
+        -------
+        :class:`Thread`
+            The newly created thread.
+        """
+        from .webhook.async_ import handle_message_parameters
+
+        params = handle_message_parameters(
+            content,
+            embed=embed,
+            embeds=embeds,
+            file=file,
+            files=files,
+            view=view,
+            components=components,
+            allowed_mentions=allowed_mentions,
+            stickers=stickers,
+        )
+
+        if auto_archive_duration is not None:
+            auto_archive_duration = cast(
+                "ThreadArchiveDurationLiteral", try_enum_to_int(auto_archive_duration)
+            )
+
+        files = cast("List[File]", params.files)
+
+        # TODO: This currently does two API calls, use HTTPClient.create_forum_post when it's available
+        thread_data = await self._state.http.start_thread_without_message(
+            self.id,
+            type=ChannelType.public_thread.value,
+            name=name,
+            auto_archive_duration=auto_archive_duration or self.default_auto_archive_duration,
+            rate_limit_per_user=slowmode_delay or 0,
+            reason=reason,
+        )
+        if files is not MISSING:
+            if len(files) > 10:
+                raise InvalidArgument("files parameter must be a list of up to 10 elements")
+            elif not all(isinstance(file, File) for file in files):
+                raise InvalidArgument("files parameter must be a list of File")
+
+            try:
+                if params.payload and "attachments" in params.payload:
+                    # We delete it since `send_files` already handles it
+                    del params.payload["attachments"]
+                message_data = await self._state.http.send_files(
+                    thread_data["id"], files=files, **params.payload  # type: ignore
+                )
+            finally:
+                for f in files:
+                    f.close()
+        else:
+            content = params.payload.pop("content") if params.payload else content
+            message_data = await self._state.http.send_message(
+                thread_data["id"],
+                content=content,
+                **params.payload,  # type: ignore
+            )
+
+        if view:
+            self._state.store_view(view, int(message_data["id"]))
+
+        return Thread(guild=self.guild, data=thread_data, state=self._state)
+
+    def archived_threads(
+        self,
+        *,
+        private: bool = False,
+        joined: bool = False,
+        limit: Optional[int] = 50,
+        before: Optional[Union[Snowflake, datetime.datetime]] = None,
+    ) -> ArchivedThreadIterator:
+        """Returns an :class:`~disnake.AsyncIterator` that iterates over all archived threads in the guild.
```

Comment:
``in the channel.``

Wherever this was copied from (I think TextChannel) should also be changed.

Reply:
😓😓